[18.0] [MIG] l10n_it_vat_registries

odooNextev opened the pull request.

eLBati committed:
copy l10n_it_vat_registries from 10.0 branch due to conflits with git AM

Commit:
porting l10n_it_vat_registries to 11.0

eLBati committed:
IMP l10n_it_vat_registries README (#479) and tests

eLBati committed:
[MIG] l10n_it_vat_registries: Migration to 12.0

eLBati committed:
IMP l10n_it_vat_registries translations

primes2h committed:
Translated using Weblate (Italian)

Currently translated at 75.0% (45 of 60 strings)

Translation: l10n-italy-12.0/l10n-italy-12.0-l10n_it_vat_registries
Translate-URL: https://translation.odoo-community.org/projects/l10n-italy-12-0/l10n-italy-12-0-l10n_it_vat_registries/it/

SimoRubi committed:
[IMP] l10n_it_vat_registries: Allow to print reports with no account moves (#1331)

alessandrocamilli committed:
[FIX][l10n_it_vat_registries]
Negative tax line in supplier invoice, must be negative in vat register line

primes2h committed:
Translated using Weblate (Italian)

Currently translated at 96.6% (57 of 59 strings)

Translation: l10n-italy-12.0/l10n-italy-12.0-l10n_it_vat_registries
Translate-URL: https://translation.odoo-community.org/projects/l10n-italy-12-0/l10n-italy-12-0-l10n_it_vat_registries/it/

eLBati committed:
IMP l10n_it_vat_registries printing VAT reg name if present

alessandrocamilli committed:
Translated using Weblate (Italian)

Currently translated at 98.3% (58 of 59 strings)

Translation: l10n-italy-12.0/l10n-italy-12.0-l10n_it_vat_registries
Translate-URL: https://translation.odoo-community.org/projects/l10n-italy-12-0/l10n-italy-12-0-l10n_it_vat_registries/it/

Commit:
merge PR 1264 1717 1728 1737 1739

sergiocorato committed:
[12.0][FIX] remove PO ref from supplier invoice reference printed in vat registry

sergiocorato committed:
Translated using Weblate (Italian)

Currently translated at 98.3% (58 of 59 strings)

Translation: l10n-italy-12.0/l10n-italy-12.0-l10n_it_vat_registries
Translate-URL: https://translation.odoo-community.org/projects/l10n-italy-12-0/l10n-italy-12-0-l10n_it_vat_registries/it/

OpenCode committed:
[FIX] l10n_it_vat_registries - Test on midnight

SilvioGregorini committed:
[UPD][l10n_it_vat_registries] Lesser update to '_get_tax_lines' method to make it more customizable

fredzamoabg committed:
[IMP] l10n_it_vat_registries: black, isort, prettier

fredzamoabg committed:
[MIG] l10n_it_vat_registries: Migration to 14.0

TheMule71 committed:
FIX l10n_it_vat_registries: use consistent order in report summary

Commit:
[FIX][l10n_it_vat_registries] fix issue https://github.com/OCA/l10n-italy/issues/2426 fix : 'account.move' object has no attribute 'amount' because in odoo 14.0 the field change name

SimoRubi committed:
[FIX] l10n_it_vat_registries: Remove report from move invoice context menu

Borruso committed:
[IMP] l10n_it_vat_registries: print purchase vat registries with ref in INV. NUM.

TheMule71 committed:
[14.0][FIX] l10n_it_vat_registries: fix bill number in report

eLBati committed:
Translated using Weblate (Italian)

Currently translated at 98.3% (58 of 59 strings)

Translation: l10n-italy-14.0/l10n-italy-14.0-l10n_it_vat_registries
Translate-URL: https://translation.odoo-community.org/projects/l10n-italy-14-0/l10n-italy-14-0-l10n_it_vat_registries/it/

sysadminmatmoz committed:
Translated using Weblate (Slovenian)

Currently translated at 54.2% (32 of 59 strings)

Translation: l10n-italy-14.0/l10n-italy-14.0-l10n_it_vat_registries
Translate-URL: https://translation.odoo-community.org/projects/l10n-italy-14-0/l10n-italy-14-0-l10n_it_vat_registries/sl/

mymage committed:
Translated using Weblate (Italian)

Currently translated at 100.0% (59 of 59 strings)

Translation: l10n-italy-14.0/l10n-italy-14.0-l10n_it_vat_registries
Translate-URL: https://translation.odoo-community.org/projects/l10n-italy-14-0/l10n-italy-14-0-l10n_it_vat_registries/it/

francesco-ooops committed:
Translated using Weblate (Italian)

Currently translated at 100.0% (59 of 59 strings)

Translation: l10n-italy-14.0/l10n-italy-14.0-l10n_it_vat_registries
Translate-URL: https://translation.odoo-community.org/projects/l10n-italy-14-0/l10n-italy-14-0-l10n_it_vat_registries/it/

Borruso committed:
[MIG] l10n_it_vat_registries: Migration to 16.0

Co-authored-by: SirAionTech <simone.rubino@aion-tech.it>

mymage committed:
Translated using Weblate (Italian)

Currently translated at 100.0% (59 of 59 strings)

Translation: l10n-italy-16.0/l10n-italy-16.0-l10n_it_vat_registries
Translate-URL: https://translation.odoo-community.org/projects/l10n-italy-16-0/l10n-italy-16-0-l10n_it_vat_registries/it/

SirAionTech committed:
Updated from template

SirAionTech committed:
[REF] l10n_it_vat_registries: Update repo template

SirAionTech committed:
[REF] l10n_it_vat_registries: Reformatted after template update

SirAionTech committed:
Updated from template

eLBati committed:
IMP l10n_it_vat_registries: ordering option

eLBati committed:
IMP l10n_it_vat_registries allowing to show full contact address in report

mymage committed:
Translated using Weblate (Italian)

Currently translated at 100.0% (62 of 62 strings)

Translation: l10n-italy-16.0/l10n-italy-16.0-l10n_it_vat_registries
Translate-URL: https://translation.odoo-community.org/projects/l10n-italy-16-0/l10n-italy-16-0-l10n_it_vat_registries/it/

eLBati committed:
[FIX] l10n_it_vat_registries: supplier refund sign

Steps:

 - Create a supplier refund
 - Print VAT registry

Observed: invoice total is negative but tax amounts are positive

Desired: tax amounts are negative

mymage committed:
Translated using Weblate (Italian)

Currently translated at 100.0% (63 of 63 strings)

Translation: l10n-italy-16.0/l10n-italy-16.0-l10n_it_vat_registries
Translate-URL: https://translation.odoo-community.org/projects/l10n-italy-16-0/l10n-italy-16-0-l10n_it_vat_registries/it/

SirAionTech committed:
[REF] l10n_it_vat_registries: Allow to override moves domain

SirAionTech committed:
[REF] l10n_it_vat_registries: Declare report columns once

matteoopenf committed:
[FIX] l10n_it_vat_registries  fix issue https://github.com/OCA/l10n-italy/issues/4612 put entry_order required to avoid crash

odooNextev committed:
[IMP] l10n_it_vat_registries: pre-commit auto fixes

odooNextev committed:
[MIG] l10n_it_vat_registries: Migration to 18.0

eLBati commented:
/ocabot migration l10n_it_vat_registries

eLBati commented:
/ocabot rebase

eLBati committed:
l10n_it_vat_registries: REF removing cee_type

eLBati commented:
/ocabot rebase

eLBati commented on test-requirements.txt:
```diff
@@ -0,0 +1 @@
+odoo-addon-l10n_it_account @ git+https://github.com/OCA/l10n-italy.git@refs/pull/4555/head#subdirectory=l10n_it_account
```
merged, si può togliere

odooNextev replied:
Fatto

odooNextev replied:
Test passati dopo aver fatto il FW del commit per rimuovere cee_type

eLBati commented:
/ocabot rebase

eLBati commented:
/ocabot rebase

Commit:
Translated using Weblate (German)

Currently translated at 28.5% (18 of 63 strings)

Translation: l10n-italy-16.0/l10n-italy-16.0-l10n_it_vat_registries
Translate-URL: https://translation.odoo-community.org/projects/l10n-italy-16-0/l10n-italy-16-0-l10n_it_vat_registries/de/

eLBati committed:
IMP l10n_it_account: _compute_totals_tax not called anymore on children taxes

monen17 commented on l10n_it_vat_registries/i18n/de.po:
```diff
@@ -0,0 +1,365 @@
+"Report-Msgid-Bugs-To: \n"
+"POT-Creation-Date: 2017-05-12 02:40+0000\n"
+"PO-Revision-Date: 2017-05-12 02:40+0000\n"
+"Last-Translator: Niki Waibel <niki.waibel@gmail.com>, 2017\n"
```
Manca https://github.com/OCA/l10n-italy/commit/1684bfeacbe31c596589db7e62e858e83dd10c5b da `16.0`.

eLBati replied:
aggiungo

monen17 replied:
:+1:

monen17 commented on l10n_it_account/models/account_tax.py:
```diff
@@ -94,6 +118,26 @@ def compute_balance(
                 0
             ]["balance"]
             balance = balance and -balance or 0
+        if debit_only is True:
```
Qui e sotto, basterebbe
```suggestion
        if debit_only:
```

eLBati replied:
refactored

monen17 replied:
:+1: questa parte non c'è più

monen17 commented on l10n_it_account/models/account_tax.py:
Le modifiche a `l10n_it_account` andrebbero in commit separati, per mantenere quanto più possibile la relazione 1 commit <-> 1 modulo modificato.

In questo caso poi vengono praticamente aggiunti solo 2 campi calcolati e potrebbe benissimo essere una feature a sé, con i suoi test e la sua issue, vedi https://github.com/OCA/l10n-italy/wiki/Team-di-sviluppo#apertura-issue.

eLBati replied:
> andrebbero in commit separati

Fatto

monen17 replied:
:+1:

monen17 commented on l10n_it_account/models/account_tax.py:
```diff
@@ -94,6 +118,26 @@ def compute_balance(
+            balance = self.env["account.move.line"].read_group(domain, ["balance"], [])[
+                0
+            ]["balance"]
+            balance = balance and -balance or 0
```
Tutti i branch di questo metodo sono molto simili, in particolare questa parte è ripetuta identica e ora aggiungiamo altre due ripetizioni, si potrebbe scriverla una volta sola?
Vedi tu se correggere anche le occorrenze che già c'erano.

eLBati replied:
refactored

monen17 replied:
:+1:

monen17 commented on l10n_it_vat_registries/wizard/print_registro_iva.py:
```diff
@@ -0,0 +1,167 @@
+        datas_form["from_date"] = wizard.from_date
+        datas_form["to_date"] = wizard.to_date
+        datas_form["journal_ids"] = [j.id for j in wizard.journal_ids]
+        if wizard._include_rc_journals():
+            datas_form["rc_journal_ids"] = [j.id for j in wizard.rc_journal_ids]
```
```suggestion
        if wizard._include_rc_journals():
            datas_form["journal_ids"].extend([j.id for j in wizard.rc_journal_ids])
```
Perché non aggiungerli a `journal_ids`? Tanto poi mi sembra vengano messi tutti insieme.

eLBati replied:
Perchè esplicito è meglio di implicito: meglio chiarire che non sono registri normali, ma dedicati a operazioni RC

monen17 replied:
:+1:

monen17 commented on l10n_it_vat_registries/readme/CREDITS.md:
```diff
@@ -1,3 +0,0 @@
-The development of this module has been financially supported by:
-
-- Odoo Italia Network
```
Se è stato fatto, non vedo perché rimuoverlo.

monen17 replied:
Perché va rimosso?
Magari visto che questa migrazione è finanziata dall'associazione si potrebbe invece aggiungere "Associazione Odoo Italia", @sergiocorato & co cosa ne pensate?

eLBati replied:
Fatto

monen17 replied:
Grazie, però hai ripristinato il file in un commit e rigenerato il README in un altro, puoi farlo nello stesso commit?

monen17 commented on l10n_it_vat_registries/views/account_view.xml:
```diff
@@ -7,7 +7,6 @@
         <field name="arch" type="xml">
             <field name="type_tax_use" position="after">
                 <field name="exclude_from_registries" />
-                <field name="cee_type" invisible="type_tax_use" />
```
Se le modifiche di questo commit sono dovute al porting alla `18.0`, andrebbero nel commit di migrazione.

eLBati replied:
No, riguardano questo https://github.com/OCA/l10n-italy/pull/4696

monen17 replied:
Ah ho capito, è un commit che è stato mergiato in `16.0` dopo che la migrazione era già stata iniziata.
Andrebbe prima del commit di migrazione, ma se è troppo difficile va bene anche qui.

monen17 commented on l10n_it_account/models/account_tax.py:
```diff
@@ -161,43 +161,19 @@ def _compute_totals_tax(self, data):
 
         tax = self.env["account.tax"].with_context(**context).browse(self.id)
         tax_name = tax._get_tax_name()
-        if not tax.children_tax_ids:
```
Come mai serve questa modifica?

eLBati replied:
Perchè a359983f4ce1c17bd4f6907d203179be4fda11f1

monen17 replied:
> Perchè [a359983](https://github.com/OCA/l10n-italy/commit/a359983f4ce1c17bd4f6907d203179be4fda11f1)

Da dove viene quel commit? In `16.0` non c'è: https://github.com/OCA/l10n-italy/blob/0a48656509df896ec599a7a9885a612536110cbd/l10n_it_account/models/account_tax.py#L164

eLBati replied:
Viene dal lavoro di porting e miglioramento del modulo

monen17 replied:
Non lo stiamo già facendo in questa PR il porting del modulo?
Se è un miglioramento, andrebbe riportato anche per le altre versioni?

monen17 commented on l10n_it_vat_registries/models/account_tax_registry.py:
```diff
@@ -33,3 +33,9 @@ class AccountTaxRegistry(models.Model):
         required=True,
     )
     show_full_contact_addess = fields.Boolean()
+    include_rc_moves = fields.Boolean(
+        string="Include reverse charge moves",
+    )
```
Questa è una nuova feature, quindi dovrebbe essere in un commit **[IMP]**, vedi https://github.com/OCA/odoo-community.org/blob/master/website/Contribution/CONTRIBUTING.rst#71commit-message.
Essendo una nuova feature, ci vorrebbero almeno i suoi test e la sua issue, vedi https://github.com/OCA/l10n-italy/wiki/Team-di-sviluppo#apertura-issue.

Non vorrei poi che interferisse con il processo di migrazione, quindi si potrebbe mettere in una PR a parte?

eLBati replied:
Non la ritengo una nuova funzionalità in quanto senza questa i registri IVA sulla 18 non funzionano

monen17 replied:
> senza questa i registri IVA sulla 18 non funzionano

Allora andrebbe nel commit di migrazione.

Poi in che senso non funzionano? Vuoi dire che se l'utente non abilita questo flag allora i registri vengono sbagliati?
Puoi aggiungere della documentazione?

eLBati replied:
> Allora andrebbe nel commit di migrazione

@monen17 però il commit di migrazione ha un altro autore

> Vuoi dire che se l'utente non abilita questo flag allora i registri vengono sbagliati?

Esatto

> Puoi aggiungere della documentazione?

Fatto in `CONFIGURE.md`

monen17 replied:
> > Allora andrebbe nel commit di migrazione
> 
> @monen17 però il commit di migrazione ha un altro autore

O ti metti come co-autore e teniamo un solo commit di migrazione (secondo me meglio), oppure lo lasci in un commit separato, ma almeno che inizi con `[MIG]`.

monen17 commented on l10n_it_vat_registries/models/vat_registry.py:
```diff
@@ -64,7 +64,6 @@ def _tax_amounts_by_tax_id(self, move, move_lines, registry_type):
         res = {}
 
         for move_line in move_lines:
-            set_cee_absolute_value = False
```
Questo commit si potrebbe mettere nel punto giusto della storia?
Visto che viene dal commit https://github.com/OCA/l10n-italy/commit/30b7c3d95afbabc5a99f42828d7bde019e96fe78 di `16.0` andrebbe prima del commit di migrazione.

eLBati replied:
ma è stato applicato dopo e ora darebbe conflitti

monen17 replied:
:+1:

monen17 commented on l10n_it_vat_registries/__manifest__.py:
```diff
@@ -0,0 +1,36 @@
+        "base_setup",
+        "account",
+        "l10n_it_account",
+        "l10n_it_edi",
```
Come mai serve questa nuova dipendenza?

eLBati replied:
Per `l10n_it_edi_is_self_invoice`

monen17 replied:
:+1:

monen17 commented on l10n_it_vat_registries/wizard/print_registro_iva.py:
```diff
@@ -0,0 +1,167 @@
+    include_rc_moves = fields.Boolean(
+        string="Include reverse charge moves",
+    )
+    rc_journal_ids = fields.Many2many(
+        "account.journal",
+        "registro_iva_rc_journals_rel",
+        "journal_id",
+        "registro_id",
+        string="Reverse charge journals",
+        help="Select journals you want retrieve reverse charge documents from",
+    )
```
```suggestion
    rc_journal_ids = fields.Many2many(
        "account.journal",
        "registro_iva_rc_journals_rel",
        "registro_id",
        "journal_id",
        string="Reverse charge journals",
        help="Select journals you want retrieve reverse charge documents from",
    )
```
Se poi potessimo anche usare l'inglese sarebbe meglio :smile:, tipo `registry_id` invece di `registro_id`.

eLBati replied:
Cioè invertiresti l'ordine ? La prima colonna non punto al record di `account.journal`?

monen17 replied:
> Cioè invertiresti l'ordine ?

Sì perchè i `Many2Many` sono definiti così, vedi https://github.com/odoo/odoo/blob/44e61c6b429aab94e4656ade5e0f0275b905210a/odoo/fields.py#L4846-L4850:
>    :param str column1: optional name of the column referring to "these" records
>        in the table ``relation``
>
>    :param str column2: optional name of the column referring to "those" records
>        in the table ``relation``

In questo caso "these" sono i registri IVA e "those" sono i registri di Odoo.

eLBati replied:
fatto

monen17 commented on l10n_it_account/models/account_tax.py:
```diff
@@ -158,46 +220,32 @@ def _compute_totals_tax(self, data):
         registry_type = data.get("registry_type", "customer")
         if data.get("journal_ids"):
             context["vat_registry_journal_ids"] = data["journal_ids"]
+            if data.get("rc_journal_ids"):
+                context["vat_registry_journal_ids"] += data["rc_journal_ids"]
```
Se usassimo solo `journal_ids`, questa modifica non servirebbe.

eLBati replied:
https://github.com/OCA/l10n-italy/pull/4664#discussion_r2063775073

monen17 replied:
ok lasciamoli pure separati.

Con gli `if` innestati in questo modo se uno seleziona solo dei `rc_journal_ids` non vengono usati, come mai?
Credo sia un comportamento quantomeno inaspettato per l'utente.

eLBati replied:
Modificato

monen17 reviewed (changes requested):
Grazie della PR!
Ho fatto solo revisione del codice.
Oltre alle cose scritte sotto, i messaggi di alcuni commit andrebbero tradotti in inglese per https://github.com/OCA/odoo-community.org/blob/master/website/Contribution/CONTRIBUTING.rst#71commit-message:
> Commit messages are in English

micheledic commented on l10n_it_account/models/account_tax.py:
```diff
@@ -158,22 +220,32 @@ def _compute_totals_tax(self, data):
             balance,
             deductible_balance,
             undeductible_balance,
+            debit_balance,
+            credit_balance,
```
Io farei una PR per la sola migrazione di vat registries senza questi due campi sul report e queste modifiche che fanno riferimento a questi due nuovi valori su tax_amounts  le farei su una PR a parte includendo tutti i moduli che sfruttano questi valori nuovi .  I commit su l10n_it_account sarebbero sempre da fare a parte visto che tutti dipendono da questi.
Ad esempio se questa MIG di vat registries viene mergiata , la PR di l10n_it_account_vat_settlement non funziona in quanto viene fatto l'unpack di meno valori nella funzione e darebbe too many values to unpack

eLBati replied:
La sola migrazione di vat registries vorebbe dire rilasciare un modulo che non funziona

monen17 replied:
Allora queste modifiche dovrebbero essere nel commit di migrazione.

eLBati replied:
Eh però https://github.com/OCA/l10n-italy/pull/4664#discussion_r2081105889

monen17 replied:
O ti metti come co-autore e teniamo un solo commit di migrazione (secondo me meglio), oppure lo lasci in un commit separato, ma almeno che inizi con `[MIG]`.

monen17 replied:
Non avevo notato che queste sono modifiche a `l10n_it_account`, quindi va bene lasciarle come `IMP` del modulo `l10n_it_account`.

A quanto ho capito questa nuova feature di `l10n_it_account` serve a far funzionare correttamente i registri IVA nella `18.0`, quindi andrebbe in una PR a parte da mergiare prima della migrazione.
In questo modo si eviterebbero problemi di dipendenza tra varie PR di migrazione (mi scrivevi che la https://github.com/OCA/l10n-italy/pull/4675 dipende da questa PR proprio a causa di queste fix).

Mi scrivevi però che ti complica la fase di sviluppo, quindi va bene lasciarlo qui, alla fine per quanto riguarda la storia dei commit è solo importante che sia in un commit separato.
Riesci però a spostarlo prima di tutti i commit di migrazione? Così dovresti anche riuscire a unire https://github.com/OCA/l10n-italy/pull/4664/commits/6678fe896d6550e5efc2664d4b6d52f6b2440f74 e https://github.com/OCA/l10n-italy/pull/4664/commits/5ac3c02f41b5a881be965a3bf4f8bbb2acf5c786 come suggerivo in https://github.com/OCA/l10n-italy/pull/4664#discussion_r2071255515.

eLBati committed:
[MIG] REF l10n_it_vat_registries for reverse charge v18:
 - show debit/credit tax totals for reverse charge operations
 - include reverse charge documents in sales VAT registry

eLBati committed:
IMP l10n_it_account: add debit_balance and credit_balance to account.tax

VBNext reviewed (approved):
Funzionale OK

Borruso reviewed (approved):
LGTM

MaurizioPellegrinet reviewed (approved):
Test funzionale: OK

monen17 commented on l10n_it_account/models/account_tax.py:
```diff
@@ -27,17 +29,7 @@ class AccountTax(models.Model):
     )
     def _compute_deductible_balance(self):
         for tax in self:
-            account_ids = (
-                tax.mapped("invoice_repartition_line_ids.account_id")
-                | tax.mapped("refund_repartition_line_ids.account_id")
-            ).ids
-            balance_regular = tax.compute_balance(
-                tax_or_base="tax", financial_type="regular", account_ids=account_ids
-            )
-            balance_refund = tax.compute_balance(
-                tax_or_base="tax", financial_type="refund", account_ids=account_ids
-            )
-            tax.deductible_balance = balance_regular + balance_refund
+            tax.deductible_balance = tax.credit_balance
```
Nel calcolo di `credit_balance` c'è anche un filtro sui conti "asset", che però qui non c'era, come mai va aggiunto?

eLBati replied:
Il `deductible_balance` è sempre comunque calcolato su registrazioni di credito IVA, quindi conti asset. Non è un'aggiunta nel senso che non cambia il risultato, è un altro modo per calcolarlo

monen17 replied:
grazie del chiarimento :smile:
:+1:

monen17 commented on l10n_it_account/models/account_tax.py:
```diff
@@ -47,21 +39,64 @@ def _compute_deductible_balance(self):
+        for tax in self:
+            accounts = tax._get_accounts_tax()
+            accounts = accounts.filtered(
+                lambda a: a.account_type.startswith("liability")
```
Qui e nell'altro `startswith`, penso sia meglio indicare esplicitamente quali sono i tipi di conto da considerare come debito/credito, come viene fatto ad esempio per i tipi di registrazioni contabili in https://github.com/odoo/odoo/blob/44e61c6b429aab94e4656ade5e0f0275b905210a/addons/account/models/account_move.py#L5524-L5533.

eLBati replied:
modificato

monen17 commented on l10n_it_account/models/account_tax.py:
```diff
@@ -158,22 +193,33 @@ def _compute_totals_tax(self, data):
+            debit_balance = -debit_balance
+            credit_balance = -credit_balance
+        if registry_type == "customer" and tax.type_tax_use == "purchase":
+            # caso reverse charge in regsitro IVA vendite
```
I commenti, come i messaggi dei commit, devono essere in inglese, vedi https://github.com/OCA/odoo-community.org/blob/master/website/Contribution/CONTRIBUTING.rst#33idioms:
> Use English variable names and write comments in English.

eLBati replied:
modificato

monen17 reviewed (changes requested):
> Oltre alle cose scritte sotto, i messaggi di alcuni commit andrebbero tradotti in inglese per https://github.com/OCA/odoo-community.org/blob/master/website/Contribution/CONTRIBUTING.rst#71commit-message:
> 
> > Commit messages are in English

Ci sono ancora dei commit in italiano, sono da scrivere in inglese:
![image](https://github.com/user-attachments/assets/2dd4e580-3df2-4d3e-beaf-f13530178c25)

eLBati commented:
> i messaggi di alcuni commit andrebbero tradotti in inglese

fatto

monen17 commented on l10n_it_vat_registries/report/report_registro_iva.xml:
```diff
@@ -299,18 +301,28 @@
+                                                />
+                                            </t>
+                                            <t t-if="registry_type == 'supplier'">
+                                                <!-- tax_code_tuple[6] is credit_balance, tax_code_tuple[4] is undeductible, which must be added to credit to get total tax amount -->
```
Grazie della spiegazione :pray:
prima o poi forse potremo avere dei nomi invece che dei numeri

monen17 commented on l10n_it_vat_registries/static/description/index.html:
```diff
@@ -475,10 +481,20 @@ <h2><a class="toc-backref" href="#toc-entry-6">Contributors</a></h2>
 <li>Alex Comba &lt;<a class="reference external" href="mailto:alex.comba&#64;agilebg.com">alex.comba&#64;agilebg.com</a>&gt;</li>
 </ul>
 </div>
+<div class="section" id="other-credits">
+<h2><a class="toc-backref" href="#toc-entry-7">Other credits</a></h2>
+<p>The development of this module has been financially supported by:</p>
+<ul class="simple">
+<li>Odoo Italia Network</li>
+<li>APS Odoo Italia</li>
+</ul>
+</div>
```
Questo andrebbe nel commit in cui ha ripristinato il file `CREDITS`.

eLBati replied:
e vabè

monen17 reviewed (changes requested):
Grazie delle modifiche!
C'è solo una modifica al codice che non mi torna: quella delle `children_tax_ids`, per il resto sono problemi di commit.

Per i problemi di commit, la struttura ideale:
- Modifiche a `l10n_it_account` in una PR separata
- In questa PR:
  1. Tutti i commit della `16.0` + pre-commit per la migrazione
  2. Commit dovuti alla migrazione di `l10n_it_vat_registries`

Visto però che questa struttura non si riesce ad applicare perché ci sono commit applicati prima/dopo, autori diversi ecc. possiamo anche lasciare tutto in questa PR con questa struttura:
1. Modifiche a `l10n_it_account`
2. Tutti i commit della `16.0` + pre-commit per la migrazione
3. Commit dovuti alla migrazione di `l10n_it_vat_registries`
4. Commit di `16.0` che sono stati applicati dopo la migrazione, e che non si riescono a mettere nel punto giusto della storia
5. Eventuali altri commit dovuti alla migrazione di `l10n_it_vat_registries` se non si riescono a schiacciare nel commit di migrazione, che però almeno inizino con `[MIG]`.

Se devi unire commit di autori diversi, mettiti come co-autore (vedi https://docs.github.com/en/pull-requests/committing-changes-to-your-project/creating-and-editing-commits/creating-a-commit-with-multiple-authors).

eLBati commented:
https://github.com/OCA/l10n-italy/pull/4664#discussion_r2081451855 forse sì ma meglio fare così ora, lasciare indietro le altre versioni e andare avanti

eLBati commented:
/ocabot merge nobump
